### PR TITLE
Stop sync

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-18024

## Description of Intent of Change(s)
PR to stop sync with tenant schema logic, update the principal clean up, flip the flag for unit tests

## Local Testing
Unit tests

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/rbac/middleware.py [hunk 1]
```diff
@@ -85,23 +82,7 @@ def get_tenant(self, model, hostname, request):
...
-                            tenant.ready = True
-                            tenant.save()
-                    except Exception as e:
-                        tenant.delete()
-                        raise e
-                else:
-                    while not tenant.ready:
-                        time.sleep(0.5)
-                        tenant.refresh_from_db()
+                tenant, created = Tenant.objects.get_or_create(schema_name=tenant_schema, defaults={"ready": True})
```
Comment (coderbydesign) [inline rbac/rbac/middleware.py:85]: Good call on having the default here, vs a keyword arg which would try to create in the case of "stuck" tenants. 👍

File: rbac/management/group/definer.py [hunk 2]
```diff
@@ -63,19 +62,15 @@ def seed_group(tenant):
...
-            return cloned_group
+        group = Group.objects.get(system=True)
+        group.name = "Custom default access"
+        group.system = False
+        group.tenant = tenant
+        group.save()
     return group
 
 
+# NOT USED ANY MORE
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply rbac/management/group/definer.py:73]: Sure

**Review by coderbydesign — APPROVED** [review]
Tested locally after the changes for the default group clone, and it seems to be good! 👍

**Review by wcmitchell — APPROVED** [review]
lgtm

**Pull request merged**